src/webui/api/torrentscontroller.cpp

```diff
@@ -815,7 +933,26 @@ void TorrentsController::addAction()
             ? Utils::String::toEnum(contentLayoutParam, BitTorrent::TorrentContentLayout::Original)
             : std::optional<BitTorrent::TorrentContentLayout> {});
 
-    const BitTorrent::AddTorrentParams addTorrentParams
+    QList<BitTorrent::DownloadPriority> filePriorities;
+    const QString filePrioritiesParam = params()[u"filePriorities"_s];
+    if (!filePrioritiesParam.isEmpty())
+    {
+        for (const QString &priorityStr : filePrioritiesParam.split(u','))
```

Comment:
You should do `filePriorities.reserve(N);` based on how many items in `filePrioritiesParam.split(u',')`